catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/catalog/CatalogApplication.java

```diff
@@ -0,0 +1,389 @@
+                req.params(":id"),
+                req.queryParams("transform")));
+
+    post(
+        "/catalog/metacard",
+        (req, res) -> {
+          try {
+            final BinaryContent content =
+                catalogService.createMetacard(req.raw(), req.queryParams("transform"));
+
+            res.status(HttpStatus.SC_OK);
+            res.body(IOUtils.toString(content.getInputStream(), StandardCharsets.UTF_8));
+            res.type(content.getMimeTypeValue());
+            return res;
+          } catch (CatalogException e) {
+            return createBadRequestResponse(res, e.getMessage());
+          }
+        });
+
+    post(
+        "/catalog/",
+        (req, res) -> {
+          if (req.contentType().startsWith("multipart/")) {
+            req.attribute(
+                ECLIPSE_MULTIPART_CONFIG,
+                new MultipartConfigElement(System.getProperty(JAVA_IO_TMPDIR)));
+
+            return addDocument(
+                res,
+                req.raw().getRequestURL(),
+                req.contentType(),
+                req.queryParams("transform"),
+                req.raw(),
+                new ByteArrayInputStream(req.bodyAsBytes()));
+          }
+
+          if (req.contentType().startsWith("text/")
+              || req.contentType().startsWith("application/")) {
+            return addDocument(
+                res,
+                req.raw().getRequestURL(),
+                req.contentType(),
+                req.queryParams("transform"),
+                null,
+                new ByteArrayInputStream(req.bodyAsBytes()));
+          }
+
+          res.status(HttpStatus.SC_NOT_FOUND);
+          return res;
+        });
+
+    put(
+        "/catalog/:id",
+        (req, res) -> {
+          if (req.contentType().startsWith("multipart/")) {
+            req.attribute(
+                ECLIPSE_MULTIPART_CONFIG,
+                new MultipartConfigElement(System.getProperty(JAVA_IO_TMPDIR)));
+
+            return updateDocument(
+                res,
+                req.params(":id"),
+                req.contentType(),
+                req.queryParams("transform"),
+                req.raw(),
+                new ByteArrayInputStream(req.bodyAsBytes()));
+          }
+
+          if (req.contentType().startsWith("text/")
+              || req.contentType().startsWith("application/")) {
+            return updateDocument(
+                res,
+                req.params(":id"),
+                req.contentType(),
+                req.queryParams("transform"),
+                null,
+                new ByteArrayInputStream(req.bodyAsBytes()));
+          }
+
+          res.status(HttpStatus.SC_NOT_FOUND);
+          return res;
+        });
+
+    delete(
+        "/catalog/:id",
+        (req, res) -> {
+          try {
+            String id = req.params(":id");
+            catalogService.deleteDocument(id);
+            return javax.ws.rs.core.Response.ok(id).build();
```

Comment:
❓ Is this a problem if you don't modify and return the `res` parameter that's passed into the endpoint?

Reply:
Oops I missed that